src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-daemon-details/rgw-daemon-details.component.ts

```diff
@@ -29,8 +28,8 @@ export class RgwDaemonDetailsComponent implements OnChanges {
 
   ngOnChanges() {
     // Get the service id of the first selected row.
-    if (this.selection.hasSelection) {
-      this.serviceId = this.selection.first().id;
+    if (typeof this.selection !== 'undefined') {
```

Comment:
Any situation where this solutions isn't enough? on the html we only validate `this.selection` to display the details.
```suggestion
    if (this.selection) {
```
Same would apply to all other occurrences.